client/src/index.jsx

```diff
@@ -3,17 +3,116 @@ import ReactDOM from 'react-dom';
 
 import axios from 'axios';
 
+import RelatedItemList from './components/RelatedItemList.jsx';
+
+
 class App extends React.Component {
     constructor(props) {
         super(props);
+        this.state = {
+            relatedItems: [],
+            relatedItemInfo: [],
+            relatedItemsCurrent: [],
+            currentArr: 0,
+            maxRelatedLength: 5,
+            itemId: 40
+        }
+
+        // click bindings
+        this.arrowClick = this.arrowClick.bind(this);
+        this.itemClick = this.itemClick.bind(this);
+    }
+
+    arrowClick(dir) {
+        if (this.state.currentArr === 0) {
+            if (dir === 'left') {
+                this.setState({
+                    currentArr: this.state.relatedItemsCurrent.length - 1
+                });
+            } else if (dir === 'right') {
+                this.setState({
+                    currentArr: this.state.currentArr + 1
+                });
+            }
+        } else if (this.state.currentArr === this.state.relatedItemsCurrent.length - 1) {
+            if (dir === 'left') {
+                this.setState({
+                    currentArr: this.state.currentArr - 1
+                });
+            } else if (dir === 'right') {
+                this.setState({
+                    currentArr: 0
+                });
+            }
+        } else {
+            if (dir === 'left') {
+                this.setState({
+                    currentArr: this.state.currentArr - 1
+                });
+            } else if (dir === 'right') {
+                this.setState({
+                    currentArr: this.state.currentArr + 1
+                });
+            }
+        }
+    }
+
+    // helper function to chunk related items array
+    chunk(array, length) {
+        let chunked = [];
+        let i = 0;
+        let n = array.length;
+
+        while (i < n) {
+            chunked.push(array.slice(i, i += length));
```

Comment:
Nice doing two things in one line. The updating of `i` and slicing the array by length.

Reply:
Also nice use of helper function!